contracts/wrappers/QuestPartner.sol

```diff
@@ -0,0 +1,356 @@
+    uint256[] public partnerQuests;
+
+    mapping(uint256 => address) public creators;
+
+    mapping(uint256 => address) public rewardTokens;
+
+    bool public killed;
+
+
+    event NewPartnerQuest(uint256 indexed questID, address indexed creator);
+
+    event AddVoterBlacklist(address account);
+
+    event RemoveVoterBlacklist(address account);
+
+    event Killed();
+
+
+    modifier onlyPartner(){
+        if(msg.sender != partner) revert Errors.CallerNotAllowed();
+        _;
+    }
+
+    modifier onlyAllowed(){
+        if(msg.sender != partner && msg.sender != owner()) revert Errors.CallerNotAllowed();
+        _;
+    }
+
+    modifier notKilled(){
+        if(killed) revert Errors.Killed();
+        _;
+    }
+
+    constructor(
+        address _board,
+        address _chest,
+        address _partner,
+        address _receiver,
+        uint256 _share
+    ){
+        if(_board == address(0)) revert Errors.ZeroAddress();
+        if(_chest == address(0)) revert Errors.ZeroAddress();
+        if(_partner == address(0)) revert Errors.ZeroAddress();
+        if(_receiver == address(0)) revert Errors.ZeroAddress();
+        if(_share == 0) revert Errors.NullAmount();
+        if(_share >= 10000) revert Errors.InvalidParameter();
+
+        board = QuestBoard(_board);
+        chest = QuestTreasureChest(_chest);
+
+        partner = _partner;
+
+        feesReceiver = _receiver;
+
+        partnerShare = _share;
+    }
+
+    function getPartnerQuests() external view returns(uint256[] memory){
+        return partnerQuests;
+    }
+
+    function getBlacklistedVoters() external view returns(address[] memory){
+        return voterBlacklist;
+    }
+
+    function createQuest(
+        address gauge,
+        address rewardToken,
+        uint48 duration,
+        uint256 objective,
+        uint256 rewardPerVote,
+        uint256 totalRewardAmount,
+        uint256 feeAmount
+    ) external notKilled nonReentrant returns(uint256) {
+        if(gauge == address(0) || rewardToken == address(0)) revert Errors.ZeroAddress();
+        if(duration == 0) revert Errors.IncorrectDuration();
+        if(rewardPerVote == 0 || totalRewardAmount == 0 || feeAmount == 0) revert Errors.NullAmount();
+        // Check for the objective is handled by the QuestBoard contract
+
+        // Verifiy the given amounts of reward token are correct
+        uint256 rewardPerPeriod = (objective * rewardPerVote) / UNIT;
+
+        if((rewardPerPeriod * duration) != totalRewardAmount) revert Errors.IncorrectTotalRewardAmount();
+
+        require(_pullTokens(rewardToken, msg.sender, totalRewardAmount + feeAmount));
+
+        uint256 newQuestId = board.createQuest(gauge, rewardToken, duration, objective, rewardPerVote, totalRewardAmount, feeAmount);
+
+        creators[newQuestId] = msg.sender;
+        rewardTokens[newQuestId] = rewardToken;
+
+        require(_sendPartnerShare(rewardToken, feeAmount));
+
+        emit NewPartnerQuest(newQuestId, msg.sender);
```

Comment:
why not emit all parameters ?
gauge, rewardToken ...

Reply:
An Event with all the parameters is already emitted by the QuestBoard contract.
Here we only emit the data relevant to the Wrapper contract, to save gas.